src/ocamldep.ml

```diff
@@ -124,17 +134,47 @@ let rules ~(ml_kind:Ml_kind.t) ~dir ~modules
       match Module.file ~dir unit ml_kind with
       | None -> Build.return []
       | Some file ->
-        let ocamldep_output = Path.extend_basename file ~suffix:".d" in
+        let ocamldep_output_path file =
+          Path.extend_basename file ~suffix:".d"
+        in
         let context = SC.context sctx in
+        let all_deps_file = Path.extend_basename file ~suffix:".all-deps" in
+        let ocamldep_output = ocamldep_output_path file in
         if not (Module.Name.Set.mem already_used unit.name) then
-          SC.add_rule sctx
-            (Build.run ~context (Ok context.ocamldep)
-               [A "-modules"; Ml_kind.flag ml_kind; Dep file]
-               ~stdout_to:ocamldep_output);
-        Build.memoize (Path.to_string ocamldep_output)
-          (Build.lines_of ocamldep_output
+          begin
+            SC.add_rule sctx
+              ( Build.run ~context (Ok context.ocamldep)
+                  [A "-modules"; Ml_kind.flag ml_kind; Dep file]
+                  ~stdout_to:ocamldep_output
+              );
+            let build_paths lines =
+              let dependencies =
+                parse_deps
+                ~dir ~file ~unit ~modules ~alias_module
+                  ~lib_interface_module ~strict:true lines
+              in
+              let dependency_file_path m =
+                let path =
+                  match Module.file ~dir m Ml_kind.Intf with
+                  | Some x -> Some x
+                  | None when Option.is_some alias_module -> None
+                  | None -> Module.file ~dir m Ml_kind.Impl
+                in
+                Option.map path ~f:ocamldep_output_path
```

Comment:
This should read the `.all-deps` files, otherwise we are not computing the transitive dependencies. Doing this change should simplify the code a bit, as the `.all-deps` no longer need to use the ocamldep syntax and can use whatever syntax we want